orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/AbstractServerGroupTask.groovy

```diff
@@ -74,7 +74,7 @@ abstract class AbstractServerGroupTask extends AbstractCloudProviderAwareTask im
       return new TaskResult(ExecutionStatus.SUCCEEDED)
     }
 
-    def taskId = kato.requestOperations(cloudProvider, [[(serverGroupAction): operation]])
+    def taskId = kato.requestOperations(operation.cloudProvider, [[(serverGroupAction): operation]])
```

Comment:
@ajordens This fixes the problem in the k8s rollback stage. `getCloudProvider(stage)` was defaulting to `aws`. Changing the clouddriver request to use the operation's `cloudProvider` makes a lot of sense to me conceptually. But it will have an impact on a lot of other stages and I'd love your eyes on it.

You can see the original problem here:
![k8s-v1 deploy problem](https://cl.ly/162Q3x0Z463t/Image%202017-11-17%20at%2011.16.25%20AM.png)

Reply:
worth noting the breaks all non-aws platforms too

Reply:
We have a slightly cleaner fix andrew is patching up in a minute

Reply:
Hah, yup ... that's an oldie right there!

I wonder if we shouldn't at least visibly log when a `cloudProvider` hasn't been specified and we're falling back to `aws`.